optuna/integration/pytorch_lightning.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import absolute_import
+
+import optuna
+
+if optuna.type_checking.TYPE_CHECKING:
+    from typing import Dict  # NOQA
+
+try:
+    from pytorch_lightning.callbacks import EarlyStopping
+    _available = True
+except (ImportError, SyntaxError) as e:
+    _import_error = e
+    # PyTorchLightningPruningCallback is disabled because PyTorch Lightning is not available.
+    _available = False
+    EarlyStopping = object
+
+
+class PyTorchLightningPruningCallback(EarlyStopping):
+    """PyTorch Lightning callback to prune unpromising trials.
+
+    Example:
+
+        Add a pruning callback which observes validation accuracy.
+
+        .. code::
+
+            trainer.pytorch_lightning.Trainer(
+                early_stop_callback=PyTorchLightningPruningCallback(trial, monitor='avg_val_acc'))
+
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of the
+            objective function.
+        monitor:
```

Comment:
[casual question] Are there chances to fire this callback other than `training_step` and `validation_end`?

Reply:
To be precise, it's fired once every epoch. I've only looked into their source code since I couldn't find documentations describing it. Basically, what the trainer does is that it keeps a dict of logs. The log is overwritten after each training batch and after each validation run. So metrics from the last of them that's run is stored in the log, then passed to this callback.

Reply:
> it's fired once every epoch.

You're right. I got confused a bit.



> So metrics from the last of them that's run is stored in the log, then passed to this callback.

As to this, I don't think "avg_val_acc" valid according to the below block though I'm not sure...
https://github.com/williamFalcon/pytorch-lightning/blob/e2cabb03ba86c4dc0271a4531bce7bf2dff53e94/pytorch_lightning/callbacks/pt_callbacks.py#L21-L25